### PR TITLE
[BT-4262-build-image] Build the testtools image in this Pipeline

### PR DESCRIPTION
**Pull request by findgriffin**
### Description

The testtools image is not currently built in any of our Pipelines, so add it to the JVM driver Pipeline.

### Motivation and context

I added the JDK to the image so that we could run JVM / AWS Lambda platform test.

### How was the change tested?

```
david@mac % fly validate-pipeline -c concourse/pipeline.yml
looks good
```

### Change types

* - [x] Bug fix (non-breaking change that fixes an issue)
* - [ ] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [x] My code follows the code style of this project.
* - [ ] My change requires a change to Fauna documentation.
* - [ ] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: concourse/pipeline.yml [hunk 1]
```diff
@@ -70,6 +97,21 @@ jobs:
...
+    plan:
+      - get: testtools-deps-repo
+        trigger: true
+
+      - put: build-and-publish-testtools-image
+        inputs:
+          - testtools-deps-repo
+        params:
+          dir: testtools-deps-repo/fauna-driver-platform-tests/
+          dockerfile: testtools-deps-repo/fauna-driver-platform-tests/Dockerfile.dotnet
```
Comment (adambollen) [inline concourse/pipeline.yml:111]:
```suggestion
          dockerfile: testtools-deps-repo/fauna-driver-platform-tests/Dockerfile
```

**Review by adambollen — CHANGES_REQUESTED** [review]
Just one copy/paste leftover, but this PR looks like it includes the changes from #184 - probably best to keep them separate.

File: concourse/pipeline.yml [hunk 2]
```diff
@@ -49,6 +69,13 @@ resources:
       aws_access_key_id: ((prod-images-aws-access-key-id))
       aws_secret_access_key: ((prod-images-aws-secret-key))
       aws_region: us-east-2
+  - name: build-and-publish-testtools-image
+    type: buildx-resource
+    source:
+      repo: ((prod-images-repo-url))/devex-dx-driver-platform-tests
```
Comment (adambollen) [inline concourse/pipeline.yml:75]: Double-check me on this, but I think the ECR repo for the general image is `-drivers-`:
```suggestion
      repo: ((prod-images-repo-url))/devex-dx-drivers-platform-tests
```

**Review by adambollen — COMMENTED** [review]
Sorry, one more tiny nit that I noticed.

File: concourse/pipeline.yml (hunk 2 above)
Comment (findgriffin) [reply concourse/pipeline.yml:75]: It's `devex-dx-drivers-platform-tests` and `devex-dotnet-driver-platform-tests`

😿

**Review by adambollen — APPROVED** [review]

**Pull request merged**